File: iosApp/iosApp/Feature/Breeds/Components/BreedsErrorContent.swift
import SwiftUI

struct BreedsErrorContent: View {
    let message: String

    var body: some View {
        VStack(spacing: 0) {
            Text("😿")
                .font(.system(size: 48))

            Spacer().frame(height: 12)

            Text("Something went wrong")
                .font(.headline)
                .fontWeight(.bold)

            Spacer().frame(height: 6)

            Text(message)
                .font(.footnote)
        }
        .multilineTextAlignment(.center)
        .padding(32)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

File: androidApp/src/main/kotlin/io/posa/feature/breeds/component/BreedsErrorContent.kt
package io.posa.feature.breeds.component

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp


@Composable
internal fun BreedsErrorContent(error: Throwable) {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .padding(32.dp)
            .testTag("breeds:error"),
        contentAlignment = Alignment.Center,
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(text = "😿", style = MaterialTheme.typography.displayMedium)
            Spacer(modifier = Modifier.height(12.dp))
            Text(
                text = "Something went wrong",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
            )
            Spacer(modifier = Modifier.height(6.dp))
            Text(
                text = error.message ?: "Unknown error",
                style = MaterialTheme.typography.bodySmall,
            )
        }
    }
}